fix mismatch sub dtype

cccclai opened the pull request:
Differential Revision: D80312352

shewu-quic reviewed (commented):
Thank you very much. I hadn't noticed this pass. Could you let me know when it was applied?

cccclai commented:
> Thank you very much. I hadn't noticed this pass. Could you let me know when it was applied?

It's running during to_edge stage https://github.com/pytorch/executorch/blob/ecb639a6c22f8cac6a779c976a668f4f318c2edc/exir/program/_program.py#L684

cccclai committed:
fix mismatch sub dtype (#13447)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/13447

Reviewed By: SS-JIA, digantdesai

Differential Revision: D80312352
